Source/AwsCommonRuntimeKit/http/HttpClientConnectionManager.swift

```diff
@@ -16,30 +17,38 @@ public class HttpClientConnectionManager {
         self.options = options
         self.allocator = allocator
         let shutDownPtr: UnsafeMutablePointer<ShutDownCallbackOptions>? = fromOptionalPointer(ptr: options.shutDownOptions)
+        //Todo: add new http2 options as properties in HttpClientConnectionOptions
         var mgrOptions = aws_http_connection_manager_options(bootstrap: options.clientBootstrap.rawValue,
-                                                             initial_window_size: options.initialWindowSize,
-                                                             socket_options: options.socketOptions.rawValue,
-                                                             tls_connection_options: options.tlsOptions?.rawValue,
-                                                             monitoring_options: options.monitoringOptions?.rawValue,
-                                                             host: options.hostName.awsByteCursor,
-                                                             port: options.port,
-                                                             proxy_options: options.proxyOptions?.rawValue,
-                                                             proxy_ev_settings: options.proxyEnvSettings?.rawValue,
-                                                             max_connections: options.maxConnections,
-                                                             shutdown_complete_user_data: shutDownPtr,
-                                                             shutdown_complete_callback: { (userData) in
-                                                                guard let userData = userData else {
-                                                                    return
-                                                                }
+                initial_window_size: options.initialWindowSize,
+                socket_options: options.socketOptions.rawValue,
```

Comment:
we could make submodule updates painless by zeroing out options-structs, before setting each member. That's the way they're intended to be used

Using the Swift-generated initializer functions like this means things break whenever we take an update and members get added

don't make the change in this PR. let's do that change in its own PR, because it will be a LOT of changed lines

Reply:
This is a good idea. We can fix it keeping the future changes in mind w.r.t ABI compatibility.